templates/todo/common/infra/bicep/app/api-container-app.bicep

```diff
@@ -2,25 +2,47 @@ param name string
 param location string = resourceGroup().location
 param tags object = {}
 
+param identityName string
 param applicationInsightsName string
 param containerAppsEnvironmentName string
 param containerRegistryName string
-param imageName string = ''
 param keyVaultName string
 param serviceName string = 'api'
 param corsAcaUrl string
+param exists bool
 
-module app '../../../../../common/infra/bicep/core/host/container-app.bicep' = {
-  name: '${serviceName}-container-app-module'
+resource apiIdentity 'Microsoft.ManagedIdentity/userAssignedIdentities@2023-01-31' = {
+  name: identityName
+  location: location
+}
+
+// Give the API access to KeyVault
+module apiKeyVaultAccess '../../../../../common/infra/bicep/core/security/keyvault-access.bicep' = {
+  name: 'api-keyvault-access'
+  params: {
+    keyVaultName: keyVaultName
+    principalId: apiIdentity.properties.principalId
+  }
+}
+
+module app '../../../../../common/infra/bicep/core/host/container-app-upsert.bicep' = {
+  name: '${serviceName}-container-app'
+  dependsOn: [ apiKeyVaultAccess ]
   params: {
     name: name
     location: location
     tags: union(tags, { 'azd-service-name': serviceName })
+    identityName: apiIdentity.name
+    exists: exists
     containerAppsEnvironmentName: containerAppsEnvironmentName
     containerRegistryName: containerRegistryName
     containerCpuCoreCount: '1.0'
     containerMemory: '2.0Gi'
     env: [
+      {
+        name: 'AZURE_CLIENT_ID'
+        value: apiIdentity.properties.clientId
+      }
```

Comment:
Since the app is now using user assigned managed identity we need to set the `AZURE_CLIENT_ID`  of the managed identity to use.